### DISCUSSION TITLE
Does this firmware have support for GD32F303 RET6 CPUs?

### DISCUSSION THREAD

**Discussion by sinkingfeeling** [mriscoc/Ender3V2S1#235] [General]
Does this firmware have support for GD32F303 RET6 CPUs? As we know Creality swapped their chips without telling anyone and Marlin dev team have been silent on support for new CPUs.

**Comment by mriscoc** [comment]
GD chips has licensing issues, that's why there is no official support from Marlin team. Because this firmware is a fork of Marlin, we also don't have official support. 

UPDATE: Said that, it is possible to make this firmware to work with that chip. Some users report that they successfully flashed the regular Professional firmware in GD32 boards. Seems that the tricky part is to have a proper SD card with the correct format. Some users got a soft brick board but they could un-brick them by re-flashing the original Creality firmware and trying the update again.

**Comment by mriscoc** [comment]
UPDATE: Just try with regular firmware releases.

This is an experimental version that could be compatible with GD SoC, I don't have any board to verify it, so it is totally up to you if you want to test.

~~[Experimental-Ender3V2-422-BLTUBL-MPC-20221130.zip]()~~

Please report if it works for you.

## **Disclaimer**
THIS FIRMWARE AND ALL OTHER FILES IN THE DOWNLOAD ARE PROVIDED FREE OF CHARGE WITH NO WARRANTY OR GUARANTEE. SUPPORT IS NOT INCLUDED JUST BECAUSE YOU DOWNLOADED THE FIRMWARE. WE ARE NOT LIABLE FOR ANY DAMAGE TO YOUR PRINTER, PERSON, OR ANY OTHER PROPERTY DUE TO THE USE OF THIS FIRMWARE. IF YOU DO NOT AGREE TO THESE TERMS THEN DO NOT USE THE FIRMWARE.

**Reply by Mysteoa** [reply]
I had previously tried to put Ender3V2-422-BLTUBL on my V2 Neo, and it's getting stuck on the Creality logo. That was before I found out it has GD32. I tried this experimental build and I also compiled some maple version on my own, but it's still not getting past the Creality logo.

**Reply by Mumtaz1977** [reply]
I have E3V2 with 4.2.2 board with GD32F303RET6 SOC and I was able to upload Mriscoc firmware and it works. I don't know where the difference is. A few times during the firmware uploads I also ran into the fact that only the background of the display was lit, or only the Creality logo appeared and nothing else. At that time, I restored the factory 1.0.2, then 1.0.4 firmware and then tried the newer JYersUI or Mriscoc firmware again.

**Reply by mriscoc** [reply]
Maybe if the board is connected to a host as Octoprint or Pronterface, an error message could be found.

**Reply by Mysteoa** [reply]
I have tried previously with Ocotpint that I have it running on a phone. It's not able to connect to the printer. 

I just tried with Pronterface and this is the error get "Update failed, No SD/TF-Card or error". There is an SD card in the printer, and it doesn't look to be corrupted, as it is working fine on my pc. I would try to get a new one on, tomorrow or on Monday. From that error code message, it looks like the SD port doesn't work, which would explain why I can't get any other FW to install.

**Reply by MarkforU** [reply]
Had this issue with Octoprint, there's a initiate card, easy way to find out if my SD card slot was working was turn printer off, unplug USB cable, turn printer on with your card in and see if you can now read your file's on Card. I thought my USB port was bad also...then I realized there was a initiate media in the screen menu. Now, I don't run off SD card and don't have cards on either Machine and I'm using Octoprint from a computer to a computer rather than Pi box, I also use phone with Octoprint without issues. Too much information to write to a card while I use Octoprint, so I don't print from a SD card.

**Reply by MarkforU** [reply]
Also make sure permissions are allowed through firewall was another issue running Octoprint off of computers in order for my Octoprint phone app to work...think firewall was my biggest issue, now I don't know about Raspberry Pi if you're running from it...I only know what I ran into with 2 machines running from a computer using Python network.

**Reply by Mysteoa** [reply]
I think you are talking about a different thing. I'm not able to see what is on the SD card when the display doesn't go pass the Creality logo. I don't have a problem connecting to Octopring on my phone, but Octoprint is not able to connect to the printer that is connected with usb. I was using it like this previously, I know it was working. I'm only able to get some error msg from the printer when I connect directly to my PC and use Pronterface.

**Reply by MarkforU** [reply]
Sorry, my mistake. I gotcha now. Dang, 32gig card I use to load firmware, takes just a moment and then you're up and running normally. Must be a slot or memory card issue if you can't get firmware to load up and not get off the modded Creality screen with the new firmware build box on the screen. Wish you luck on that one...

**Reply by Mysteoa** [reply]
I manged to fix it. It was most likely a problem with the original SD card or becouse I named the FW "Please work". It was just hard to find a SD card smaller than 32gb.

**Reply by ghost** [reply]
I think the flashing with this chip is just flaky. I have this bin working now (thank you btw) but it took several flash attempts. I've had the same experience with the official bin as well.

**Reply by big-griff** [reply]
> I manged to fix it. It was most likely a problem with the original SD card or becouse I named the FW "Please work". It was just hard to find a SD card smaller than 32gb.

for anyone having issues with this in the future, you can shrink the partition of a 32gb microsd to 4gb, format it to FAT32 4096b and name your .bin a 4-digit filename for consistent flashing
the microsd that comes with creality printers is notoriously unreliable and will fail to flash despite appearing to function normally

**Comment by AlexeyPechnikov** [comment]
@mriscoc The firmware loaded fast and without any problems on the stock Ender 3 v2 using stock 8 GB micro-sd card:
```
Motherboard 4.2.2
Processor GD32F303
Stepper drivers t8 = tmc2208
Firmware version 1.0.5
```
But it does not print stock g-code file - just housing the hot end above the plate center and waiting two minutes and show that the prin is completed. There are no errors or notices. Maybe does it need CR Touch and filament runout sensors attached?

**Reply by mriscoc** [reply]
Post a capture of the info page and verify that it matches your hardware.

**Reply by AlexeyPechnikov** [reply]
Hmm, I don’t understand how is it possible to verify when we have firmware build date only on the info page. All that I know about the firmware is “This is an experimental "Maple" version that could be compatible with GD SoC” as you posted above. The info page and the printer motherboard photos attached below:

![FC5F26A5-5DFF-46D4-8618-58F925C806B9](https://user-images.githubusercontent.com/7342379/210271275-1dd4beef-fbc0-4fa5-9c4e-77c346b77e12.jpeg)
![8505CF11-8730-4377-B162-A530458F136A](https://user-images.githubusercontent.com/7342379/210271331-fedab34b-4b61-41c2-ae0c-9ee13a6cf790.jpeg)

**Reply by mriscoc** [reply]
ok, that firmware [Experimental-Ender3V2-422-BLTUBL-MPC-20221130.zip]() is only for printers with BLT (CR/3D/BLTouch)

**Reply by AlexeyPechnikov** [reply]
No problem, I can install CR Touch to test it. I didn’t install it yet just to test the stock firmware which doesn’t support CR touch. I see there are a lot of steps required and even some meshed should be imported into slicer software but is it possible to miss all the steps and just connect the sensor and print the stock g-code file to check? For now I don’t understand why your firmware requires different table surface interpolations and how should we define all the parameters and how to use the measurements.

**Comment by AlexeyPechnikov** [comment]
@mriscoc The CR touch sensor moves the probe up and dows at the printer loading time but the firmware crashes nozzle into the bed glass. CR touch sensor is not active all the time when homing performing. Probably the firmware misses as CR touch support as print possibility but it's flashing from sd-card and shows menus.

**Reply by mriscoc** [reply]
That issue was already solved, please search on closed issues or open a new one to not polute this issue with off topic post.

**Comment by mriscoc** [comment]


**Reply by big-griff** [reply]
on my machine it works but the steppers do not work correctly and will crash

**Comment by sammcj** [comment]
For anyone that's searching and finds this discussion just FYI works fine with my GD32F303, obviously YMMV etc...

![image](https://user-images.githubusercontent.com/862951/218287909-82b21ad0-0f62-451d-bdd2-72d5cada956d.png)

![IMG_8051](https://user-images.githubusercontent.com/862951/218296945-f14115f3-4e18-45f7-9aca-0fba5f200e27.jpeg)

**Reply by kp757200** [reply]
Hi, is this the experimental firmware or regular firmware? I saw post with experimental firmware for BLtouch but looks like that link is broken. Thank you

**Reply by mriscoc** [reply]
Currently, use the regular firmware.

**Comment by bagsofcole** [comment]
This FW is working fine for me too with the GD Chip

**Comment by jgemberton** [comment]
Trying to compile my own as I am upgrading my (4.2.2 board with GD chip) 3v2 with a Sprite Pro and a CR touch. Should I still use Auto Build Marlin or is there something else that will work better with this board/chip combo?

**Reply by mriscoc** [reply]
Officially, Marlin doesn't support GD SoC's due to licence infringements with STM, but several users report that the firmware works with GD chips.

Just some comments above: https://github.com/mriscoc/Ender3V2S1/discussions/235#discussioncomment-3193842

**Comment by Yasamal33** [comment]
+1 to GD chip owners (Ender 3 v2 4.2.2 board).
The main thing is that it's is a lot naughty regarding to SD card within firmware flash. I tried several ones until I found suitable one.

**Comment by VitaliyKrym** [comment]
Can you describe step by step what you did to make it work. What firmware you use? What SD size you used? I have V2 NEO BLtouch, end sensor with Octoprint. 4.2.2 GD32F303 RET6.

**Reply by Mysteoa** [reply]
In mine case for my Neo, I used 16gb TeamGroup SD card. Formatted to FAT32 using 4k unit size. I put the FW on the card and insert it in the printer while it is off. Then I turn on the printer and waited at least 5 min. After that, I turn it off and removed the card. 

Next step is to put the Display FW on the SD according to the model Screen you have. Insert it in the screen SD slot, and turn on the printer and wait for the process to finish. Shutdown the printer and removed the SD Card.

Now your printer should be working using Professional FW. If it doesn't work, try again with the printer FW. Make sure you rename the FW file after each try, because the Printer remembers the name of the last FW file, and It will not use it again. The maximum size card you can use is 32GB. You can use a bigger card, but you need to partition the SD card with a volume smaller than 32GB. I originally tried with the provided SD card, but that card is JUNK and it failed each time. I can say that I helped more than 5 people that were having failed attempts due to provided JUNK card.

**Reply by VitaliyKrym** [reply]
What firmware you used? File name? Did you do any configuration check?

**Reply by Mysteoa** [reply]
The one that is provided here for 4.2.2 with BL touch. File name doesn't matter, but you need to change it on each try. What config check?

**Reply by VitaliyKrym** [reply]
Config change. My bad. I mean what file you use from the list?

**Comment by VitaliyKrym** [comment]
It worked!!! Ender 3 V2 NEO BL 4.2.2. GD CPU. Had issue with SD card. Finally, I made 8Gb partition on FAKED SanDisk Extreem 128GB microsd from eBay.

**Reply by Mysteoa** [reply]
Good job. Like one of the upper comments mentioned, apparently it is really picky for an SD card. I was in your situation and I thought I bricked the printer. I even order a replacement 4.2.7 board. A day before it arrived, I managed to flash the FW using a brand new SD card.

**Reply by VitaliyKrym** [reply]
Before, to upgrade to 1.1.7 after filament sensor installation I tried many SDs but old 4Gb from trash worked. Now as I use this card in RPi, I had to look for another one. Do I need for change any settings after upgrade?

**Reply by Mysteoa** [reply]
Yes, by default the FW is compiled with Ended 3 V2 Physical measurements in mind, not Neo. It does have different X/Y and probe offsets. You need to calibrate your bed size, min/max X/Y and probe offset. You should read the Wiki here - https://github.com/mriscoc/Ender3V2S1/wiki/Calibration-Guides . You should also run bed and hotend MPC Tune.

Furthermore, you can probably find those calibration values in a post here, but it would be best to find them on your own and just use those as a reference.

**Reply by VitaliyKrym** [reply]
For some reason LCD firmware update does not do anything (visually). It updates and shows "Update done" but I still have old stile screen. Any suggestions? It is DACAI.

**Reply by mriscoc** [reply]
For DACAI there is a firmware update and also a Giadej icon set: https://github.com/mriscoc/Ender3V2S1/tree/Ender3V2S1-Released/display%20assets/Giadej%20compilation

**Reply by VitaliyKrym** [reply]
Do I need "private" folder only on SD?

**Reply by Tornadyx** [reply]
> It worked!!! Ender 3 V2 NEO BL 4.2.2. GD CPU. Had issue with SD card. Finally, I made 8Gb partition on FAKED SanDisk Extreem 128GB microsd from eBay.

Hey could u please tell me which version u used and what marlin build? (Since you're the only one reporting success on a v2 neo) Im trying to get mine to work but im unsuccessful so far. I also somehow can't build with the maple enviroment, idk if the mistake is there...

**Comment by VitaliyKrym** [comment]
What settings should be set for V2 NEO? Can anybody share print screens or just numbers?

**Comment by Tornadyx** [comment]
Hey Alexey thanks for your message, I actually got it to work this morning. Also, after flashing the screen with the custom TJC set from mriscoc everything works fine so far, bed tramming works, mesh levelling too. We'll see for how long but for now its great.

________________________________
Von: Alexey Pechnikov ***@***.***>
Gesendet: Dienstag, 11. Februar 2025 06:30
An: mriscoc/Ender3V2S1 ***@***.***>
Cc: Tornadyx ***@***.***>; Mention ***@***.***>
Betreff: Re: [mriscoc/Ender3V2S1] Does this firmware have support for GD32F303 RET6 CPUs? (Discussion #235)


@Tornadyx<https://github.com/Tornadyx> It is not actual anymore because free mriscoc software seems broken for about a year and tramming fails 9 times out of 10 and after that nothing works in the firmware. Maybe the paid version works I didn’t try but the free one is not for me and I found lots of reports about the same problem. Try other software like Klipper which has very simple firmware for a printer and you have a good chance it can work on your GB motherboard.

**Reply by AlexeyPechnikov** [reply]
@Tornadyx Nice to hear! For me, it was nice before I trammed my bed to check (do not actually need because I have a second Z axis with synchrobelt) and from that time the firmware often halted at random times. My CR Touch works nicely with Klipper so it is some trouble in the firmware but it worked smoothly before the tramming.

**Comment by VitaliyKrym** [comment]
I have different issue with tramming and leveling. After perfect aliment (according to software) I have left side if bed perfect offset. Same time on right side nozzle is much higher. Have to do manual leveling.
